### DISCUSSION TITLE
Cannot intercept AltChar event

### DISCUSSION THREAD

**Discussion by smallB007** [gyscos/cursive#715] [General]
Hi,
I've tried to distinguish between char and altchar events by doing:
```
let convert = |ev: &cursive::event::Event| match ev {
        Event::AltChar(_) => {
            eprintln!("Alt char");
            return false;
        }
        Event::Char(_) => {
            eprintln!("char");
            return true;
        }
        _ => {
            eprintln!("Not a char");
            return false;
        }
    };
```
Unfortunately, altchar is seen as Char(_) and I wasn't able to react on AltChar event. Could you please advice me how am I suppose to do it?
Thank you

**Comment by gyscos** [comment]
Hi,

This looks like the correct way to do it. If you don't see alt events, it's possible your terminal emulator (or an intermediary, like screen/tmux) is not properly sending these events to the application at all.

You can try running the `key_codes` example to see what is received when you press alt+letter.

It's also possible the backend is not properly reading these. What os are you running? What cursive backend?

**Reply by smallB007** [reply]
Hi,
I've followed your suggestion. The results are:
Shift+char generates char event
Alt+char generetes Esc and char events

I'm using ubuntu, cursive backend is default.

**Reply by massimo79m** [reply]
I don't know on ubuntu, but on macos the default backend is curses, that has some problems with CTRL/ALT. 
Try to force another backend with 

[dependencies.cursive]
version = "0.20.0"
default-features = false
features = ["crossterm-backend"]
